src/main/java/org/candlepin/service/exception/user/UserMissingException.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2009 - 2023 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.service.exception.user;
+
+
+/**
+ * The UserMissingException is used to reject a user validation action.
+ */
+public class UserMissingException extends UserServiceException {
```

Comment:
For now, having both a UserInvalidException and UserMissingException is okay. Today, Adapters **does** distinguish between invalid users (users with incorrect login+password combos) and users that do not exist (login was not found) for certain calls. However, I wonder if we should be returning UserInvalidException in either case, for security purposes.

Reply:
This is an interesting one. I guess the question comes down to whether or not Candlepin ever needs to care about such a differentiation and act accordingly. If not, then the adapters throwing a single "user not found" exception is probably okay. Otherwise, it may make sense to leave that up to CP to kick back the same exception in both paths.

But to the larger point, I agree that we could be more limiting on the info we give to users via error messages.